Verify the kustimze v5 breaking changes (#3251)
kubernetes-sigs/kubebuilder: docs/book/src/plugins/kustomize-v2-alpha.md

note: We bumped kustomize v5 on kustomize/v2-alpha which is used by go/v4-alpha.
Below changes may affect your current kubernetes operator. So please take a look at it.
Remove deprecated alpha cfg and fn commands
Duplicate fields in Kustomization files will now result in an error
Other than that they have fixed remote urls, Drop support for a very old, legacy style of patches, Drop support for using gh: as a host, Remove a potential build-time side-effect in ConfigMapGenerator and SecretGenerator, etc.
For detail information kindly take a look on [this](https://github.com/kubernetes-sigs/kustomize/releases) page

> # Kustomize v2-alpha

Reviewer: Hi @Sajiyah-Salat,

Thank you for looking on this one.  What we actually need to do is:
- Change here https://github.com/kubernetes-sigs/kubebuilder/blob/bb2eb918ccdd7797a2e7482718bebf3433b3e3e0/docs/book/src/plugins/kustomize-v2-alpha.md#when-to-use-it to say that is 5x instead of 4.x
- Update the section `When to use it` this plugin based on the changes in the release notes from 3.x to 5.x into the section: https://github.com/kubernetes-sigs/kubebuilder/blob/bb2eb918ccdd7797a2e7482718bebf3433b3e3e0/docs/book/src/plugins/kustomize-v2-alpha.md#when-to-use-it
- Add the release notes from 4.x to 5.x under the Future references section: https://github.com/kubernetes-sigs/kubebuilder/blob/bb2eb918ccdd7797a2e7482718bebf3433b3e3e0/docs/book/src/plugins/kustomize-v2-alpha.md#further-resources so that users can also check it out.

Could you please help us out with?
WDYT?

Author: Thanks you for this detail clarification. I will surely give it a try.

## When to use it

- If you are looking to scaffold the kustomize configuration manifests for your own language plugin

> - If you are looking for support on Apple Silicon (`darwin/arm64`). (_Before kustomize `5.x` the binary for this plataform is not provided_)

Reviewer: ```suggestion
- If you are looking for support on Apple Silicon (`darwin/arm64`). (Apple Silicon is not supported by Kustomize versions < `4.x`._)
```

> - If you are looking for to begin to try out the new syntax and features provide by kustomize v5

Reviewer: ```suggestion
- If you are looking for to begin to try out the new syntax and features provide by kustomize v4 and/or v5
```

> - If you are NOT looking to build projects which will be used on Kubernetes cluster versions < `1.22` (_The new features provides by kustomize v4 are not officially supported and might not work with kubectl < `1.22`_)

Reviewer: We need to check what k8s version is supported by kustomize v5

Author: in kustomize release note of v5 there is no mention of k8s version support. but from my research I found that: 
Kustomize v2.x.x is compatible with Kubernetes v1.9 and later.
Kustomize v3.x.x is compatible with Kubernetes v1.13 and later.
Kustomize v4.x.x is compatible with Kubernetes v1.16 and later.
Kustomize v5.x.x is compatible with Kubernetes v1.21 and later.

Author: let me know your reviews and I will try to update the file as suggested.

* Check the kustomize [plugin implementation](https://github.com/kubernetes-sigs/kubebuilder/tree/master/pkg/plugins/common/kustomize)
* Check the [kustomize documentation][kustomize-docs]
* Check the [kustomize repository][kustomize-github]

> * Check the [kustomize release note](https://github.com/kubernetes-sigs/kustomize/releases)

Reviewer: We need to add the link for kustomize v4.0.0 specific release to let people be able to check the breaking changes and also for v5.0.0 one because of the same reasons.

Author: v4: https://github.com/kubernetes-sigs/kustomize/releases/tag/kustomize%2Fv4.5.7
v5: https://github.com/kubernetes-sigs/kustomize/releases/tag/kustomize%2Fv5.0.0
I want you to just confirm that these are the links you are talking about.

Reviewer: Breaking changes are introduced usually only in MAJOR releases (see: https://semver.org/)
So, it is v4.0.0 (I think we have the link for this one can you please check in the page, if not we need to add ) and 5.0.0

Author: I tried to check the file but it is giving me 404 error

> * To know more about the changes between kustomize v4 and v5 see its [release notes][release-notes]

Reviewer: ```suggestion
* To know more about the changes between kustomize v3 and v4 see its [release notes][release-notes]
* To know more about the changes between kustomize v4 and v5 see its [release notes][release-notes]
```
Update the links

> * Also, you can compare the `config/` directory between the samples `project-v3` and `project-v3-with-kustomize-v2` to check the difference in the syntax of the manifests provided by default

Reviewer: ```suggestion
* Also, you can compare the `config/` directory between the samples `project-v3` and `project-v4` to check the difference in the syntax of the manifests provided by default
```

Reviewer: It is the project-v4 now.
